src/components/ThemeButton.tsx

```diff
@@ -0,0 +1,111 @@
+import Link from 'next/link';
+import { useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+export default function ThemeButton({
+  href,
+  children,
+  className,
+  type = 'rectangle',
+  disabled = false,
+  noBackground = false,
+  fetchOnClick = false,
+  onClick = (_e: any) => {
+    /*action*/
+  },
+  onKeyDown = (_e: any) => {
+    /*action*/
+  },
+  onKeyUp = (_e: any) => {
+    /*action*/
+  },
+  onFetchDone = async (_response: any) => {
+    /*response*/
+  },
+}: any) {
+  const [fetchStatus, setFetchStatus] = useState(false);
+
+  const pillFormat = `${type == 'pill' ? 'rounded-full ' : ''}`;
+  const noBackgroundFormat = `${
+    disabled
+      ? `cursor-not-allowed${
+          noBackground ? ' ' : ' bg-black-0 '
+        }text-black-200 `
+      : `cursor-pointer${
+          noBackground ? ' text-accent-500' : ' bg-accent-500 text-secondary'
+        }`
+  }`;
+
+  function onClickInternal(e: any) {
+    if (disabled) return;
+    if (fetchOnClick) {
+      setFetchStatus(true);
+      fetch(href).then(res => {
+        setFetchStatus(false);
+        onFetchDone(res);
+      });
+    }
+    onClick(e);
+  }
+
+  const content = (
+    <a
+      className={twMerge(
+        'px-5 py-3 flex items-center justify-center font-sans font-bold text-button',
+        pillFormat,
+        noBackgroundFormat,
+        className
```

Comment:
I remember you said you were having some issues with `twMerge` and I'm guessing it might've been spacing related? We can use the `clsx` helper inside of `twMerge` to keep it clean.

Here's an example, with the colors changes that I mentioned before.

```js
type Props = {
  variant?: 'primary' | 'secondary' | 'dark' | 'light';
  className?: string;
};

function Button({ className, variant = 'primary' }: Props) {
  return (
    <button
      className={twMerge(clsx(
        'px-5 py-3 flex items-center justify-center font-sans font-bold text-button',
        { 'bg-accent text-secondary': variant === 'primary' },
        { 'bg-secondary text-accent': variant === 'secondary' },
        { 'bg-black-100 text-secondary': variant === 'light' },
        { 'bg-black-400 text-secondary': variant === 'dark' },
        className
      ))}
    >
      Hello World
    </button>
  );
}
```

Extract that and update please